src/expr/src/vector_op/agg/general_sorted_grouper.rs

```diff
@@ -149,25 +198,33 @@ where
             ongoing = true;
             ongoing_group = v;
         }
-        Ok(EqGroups(ret))
+        Ok(EqGroups::new(ret))
     }
 
     pub fn update_and_output_with_sorted_groups_concrete(
         &mut self,
         input: &T,
+        offset: usize,
```

Comment:
Just realized another relationship between these variables:

This `offset` (and the one passed in and returned out on aggregators) is unnecessary - it can be calculated from `groups.offset`:

```
    pub fn chunk_offset(&self) -> usize {
        if self.offset > 0 { self.indices[self.offset-1] } else { 0 }
    }
```
I added this method on `EqGroups` as well as `assert(offset == groups.chunk_offset());` in both `update_and_output`, and all unit tests run happily.

Reply:
Good point. I will figure out this in later PR.